### DISCUSSION TITLE
Collider is spawned in center before go to correct position

### DISCUSSION THREAD

**Discussion by trung1-dothanh** [bevyengine/bevy#6772] [Q&A] [answered]
Hi,

I'm getting a problem when spawn Collider at the right end of window.
The problem is, when spawned, collider appears at the center of window before going to the desired position.

Video demo for problem:
https://i.imgur.com/LBuA96v.mp4

When spawn, the code just insert Collider and TransformBundle, so I'm not sure where this problem comes from.
How I can avoid this "spawn at center" problem? Thanks.

Bevy version: 0.9
Rapier version: 0.19

```rs
fn new_pole(mut command: Commands, windows: Res<Windows>) { let window = windows.get_primary().unwrap();
    ...
    command
        .spawn(Collider::cuboid(pole_width, upper_height / 2.))
        .insert(TransformBundle::from(Transform::from_xyz(
            window.width() / 2.,
            window.height() / 2. - upper_height / 2.,
            0.,
        )))
        .insert(Pole);
}

...
fn main() {
    App::new()
        .add_plugin(RapierDebugRenderPlugin::default())
        ...
        .add_system_set(
            SystemSet::new()
                .with_run_criteria(FixedTimestep::step(3.))
                .with_system(new_pole),
        )
        .run();
}

```

**Comment by DasLixou** [comment]
Instead of .insert, you should put it in a tuple together with the Collider: `commands.spawn((Collider::cuboid(..), TransformBundle::from(..));`

**Reply by trung1-dothanh** [reply]
Thank you for the suggestion.

I tried put them in a tuple but problem stays same. It is still spawned in the center first.
```rs
    command
        .spawn((
            Collider::cuboid(pole_width, upper_height / 2.),
            TransformBundle::from(Transform::from_xyz(
                window.width() / 2.,
                window.height() / 2. - upper_height / 2.,
                0.,
            )),
        ))
        .insert(Pole);
```

**Reply by DasLixou** [reply]
Then it’s probably a problem with rapier-plugin.

**Reply by trung1-dothanh** [reply]
Yeah, it seems so.

**Comment by tim-blackbird** [comment, marked as answer]
Adding the right [`RigidBody`](https://docs.rs/bevy_rapier2d/latest/bevy_rapier2d/dynamics/enum.RigidBody.html) component variant may help.

**Reply by trung1-dothanh** [reply]
Thank you. Adding `RigidBody` works.
